metric-anomaly-data-model/src/main/avro/NotificationEvent.avdl

```diff
@@ -0,0 +1,15 @@
+@namespace("org.hypertrace.alert.engine.metric.anomaly.datamodel")
+protocol NotificationEventProtocol {
+  record NotificationEvent {
+    string tenant_id;
+    long event_time_millis;
+    map<string> action_event_metadata;
```

Comment:
nit: Just (`type`, `value`, `metadata`) already part of EventRecord. But, if are using full naming everywhere, this is fine too.